### PR TITLE
Add option select component

### PR DESCRIPTION
**Pull request by andysellick**
## What
Adds the [option select component](https://govuk-finder-frontend.herokuapp.com/component-guide/option_select) from `finder-frontend`.

Code has been copied exactly from the original, apart from the following necessary changes:

- CSS class prefixes renamed from `.app-c-` to `.gem-c-`
- render component method removed from the rspec test as unnecessary
- double `::` referring to a pseudo element replaced with single `:` in the stylesheet as it failed the linter
- `add_gem_component_stylesheet("input")` added to the template as it wasn't including the styles for an input element on the 'with filter' option (see the [existing component](https://govuk-finder-frontend.herokuapp.com/component-guide/option_select/with_filter) for an example of this problem)
- removed a test to check that attributes for GA4 tracking are added correctly to the button when a `ga4-filter-indexes-added` event occurs, this should be covered by the tests for the `Ga4FinderTracker` in `finder-frontend` already
- add zero margin to the `.gem-c-option-select__heading` style, as previously reset by the CSS reset in `finder-frontend` (thanks to @MartinJJones for noticing this)

## Why
This component was recently copied into `signon` and now we have two copies of this component. Instead, make one copy here in the components gem, and after this is done, use it in the other two applications. See https://github.com/alphagov/govuk_publishing_components/blob/main/docs/moving-components-upstream-into-this-gem.md

## Visual Changes
None, apart from the new component.

Screenshot 2023-09-19 at 17 55 17

### CODE REVIEW

File: spec/javascripts/components/option-select-spec.js [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+
+    it('replaces the `span.gem-c-option-select__title` with a button', function () {
+      $element = document.createElement('div')
+      $element.innerHTML = html
+      new GOVUK.Modules.OptionSelect($element.querySelector('.gem-c-option-select')).init()
+
+      expect($($element).find('button')).toBeDefined()
+    })
+
+    it('adds GA4 event tracking to the button', function () {
```
[the remark this answers is not in the document]
Comment (AshGDS) [reply spec/javascripts/components/option-select-spec.js:156]: @andysellick Yep, it's only relevant if that tracker is enabled on the component, so can be removed here.

[the remark this answers is not in the document]
Comment (andysellick) [reply spec/javascripts/components/option-select-spec.js]: Thanks, will remove.

**Review by MartinJJones — APPROVED** [review]
Changes look good to me, nice work! 👍

**Review by MartinJJones — APPROVED** [review]

**Comment by floehopper** [comment]
@andysellick Hi there. Thank you for doing this. As it happens I've been working on doing exactly this in [this card](https://trello.com/c/qs60lCxl) as a follow-up to introducing the duplication. However, I was worried that the GA4-related code was rather coupled to finder-frontend, so I've been spiking on a refactoring of the code in finder-frontend to avoid that coupling. It's turned out to be more complicated than I expected, so I'm not sold on the approach. However, it'd be good to have a chat about all this if you have time.

**Comment by andysellick** [comment]
@floehopper ah apologies didn't mean to step on your toes. Moving this into the gem has indeed caused some interesting problems, both in [finder-frontend](https://github.com/alphagov/finder-frontend/pull/3165) and [signon](https://github.com/alphagov/signon/pull/2368), but I believe we have some possible solutions that I'm about to look into.

**Comment by floehopper** [comment]
@andysellick

> ah apologies didn't mean to step on your toes

No worries! Thanks for doing the work to extract the component, release the gem, and check the dependabot PR in the signon app. 🙏 

> Moving this into the gem has indeed caused some interesting problems, both in https://github.com/alphagov/finder-frontend/pull/3165 and https://github.com/alphagov/signon/pull/2368, but I believe we have some possible solutions that I'm about to look into.

As we discussed, in case it's useful, here's my rather scrappy attempt at moving the GA4 functionality out of the `option-select`: https://github.com/alphagov/finder-frontend/pull/3166. Let me know if you have any questions about it.

**Pull request merged**